### PR TITLE
[pkg/util/cloudproviders/gce] Add host tag for `DD_PROVIDER_KIND`

### PR DESCRIPTION
**Pull request by jennchenn**
### What does this PR do?

Add a new `provider_kind` GCE host tag with value set to `DD_PROVIDER_KIND`.

### Motivation

We set `DD_PROVIDER_KIND` for certain GKE clusters, but do not use that value anywhere. This PR adds a host tag so we can use this to understand what provider is being used.

### Describe how to test/QA your changes
1. Deploy the agent with `DD_PROVIDER_KIND` set as an environment variable
2. Verify that `provider_kind` shows up as a host tag after some time

### Possible Drawbacks / Trade-offs
At startup there is some time where host tags are not yet available and the value is set to `N/A`, even if the environment variable is set.
image

### CODE REVIEW

File: pkg/util/cloudproviders/gce/gce_tags_test.go [hunk 1]
```diff
@@ -146,3 +147,20 @@ func TestGetHostTagsWithNonDefaultTagFilters(t *testing.T) {
 	require.NoError(t, err)
 	testTags(t, tags, expectedExcludedTags)
 }
+
+func TestGetHostTagsWithProviderKind(t *testing.T) {
+	ctx := context.Background()
+	mockConfig := configmock.New(t)
+	defaultProviderKind := mockConfig.GetStringSlice("provider_kind")
```
Comment (dustmop) [inline pkg/util/cloudproviders/gce/gce_tags_test.go:154]: Is this supposed to be `GetStringSlice`? It's simply `GetString` in the non-test code.

Comment (jennchenn) [reply pkg/util/cloudproviders/gce/gce_tags_test.go]: Thanks good catch, just updated!

**Review by dustmop — APPROVED** [review]
LGTM!

File: releasenotes/notes/add-provider-kind-tag-518f76ee283bc1ff.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Add new host tag `provider_kind` from the value of `DD_PROVIDER_KIND`
+    for agents running in GCE.
```
Comment (drichards-87) [inline releasenotes/notes/add-provider-kind-tag-518f76ee283bc1ff.yaml:12]:
```suggestion
    Add a new host tag `provider_kind` using the value of `DD_PROVIDER_KIND` for Agents running in GCE.
```

**Review by drichards-87 — APPROVED** [review]
Left a suggestion from Docs and approved the PR.

**Pull request merged**